### PR TITLE
chore: Use `babel.config.cts`

### PR DESCRIPTION
**Pull request by liuxingbaoyu**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | `Fixes #1, Fixes #2` 
| Patch: Bug Fix?          |
| Major: Breaking Change?  |
| Minor: New Feature?      |
| Tests Added + Pass?      | Yes
| Documentation PR Link    | 
| Any Dependency Changes?  |
| License                  | MIT

ref: https://github.com/babel/babel/pull/15478

During the migration I found two bugs, but luckily they worked by accident.

<a href="https://gitpod.io/#https://github.com/babel/babel/pull/15511"></a>

### CODE REVIEW

File: babel.config.cts [hunk 1]
```diff
@@ -719,27 +750,31 @@ function pluginImportMetaUrl({ types: t, template }) {
...
 
           try {
             const pkg = JSON.parse(
               fs.readFileSync(
-                pathUtils.join(dir, `${source.value}/package.json`)
-              ),
-              "utf8"
+                pathUtils.join(dir, `${source.value}/package.json`),
+                "utf8"
+              )
```
Comment (liuxingbaoyu) [inline babel.config.cts:777]: Previously we accidentally passed `utf8` to `JSON.parse`.

File: babel.config.cts [hunk 2]
```diff
@@ -787,7 +822,7 @@ function getTokenTypesMapping() {
       }),
       {
         configFile: false,
-        parserOpts: { attachComments: false, plugins: ["typescript"] },
+        parserOpts: { attachComment: false, plugins: ["typescript"] },
```
Comment (liuxingbaoyu) [inline babel.config.cts:825]: There is a typo here.

File: babel.config.cts [hunk 3]
```diff
@@ -719,27 +750,31 @@ function pluginImportMetaUrl({ types: t, template }) {
...
+): PluginObject {
   return {
     visitor: {
-      "ImportDeclaration|ExportDeclaration"({ node }) {
+      "ImportDeclaration|ExportDeclaration"({
+        node,
+      }: NodePath<t.ImportDeclaration | t.ExportDeclaration>) {
+        if (!("source" in node)) return;
         const { source } = node;
-        if (!source) return;
```
Comment (nicolo-ribaudo) [inline babel.config.cts:728]: This change is causing the CI failure

**Comment by liuxingbaoyu** [comment]
This seems to be a compatibility issue with older versions of jest, I'll look into it.

**Pull request closed without merging**